modular_robot_physical/revolve2/modular_robot_physical/physical_interfaces/v1/_v1_physical_interface.py

```diff
@@ -11,68 +13,137 @@ class V1PhysicalInterface(PhysicalInterface):
     _PWM_FREQUENCY = 50
     _CENTER = 157.0
     _ANGLE60 = 64.0
+    _PINS = list(range(2, 28))
 
     _debug: bool
     _dry: bool
-    _pins: list[int]
+    _careful: bool
+    _careful_stepsize: float
+
     _gpio: pigpio.pi | None
+    _set_targets: dict[int, float]  # The targets currently actually set.
+    _targets: dict[int, float]  # The desired targets of the user.
+
+    _enabled: bool
+    _update_loop_thread: threading.Thread | None
 
-    def __init__(self, debug: bool, dry: bool, pins: list[int]) -> None:
+    def __init__(
+        self, debug: bool, dry: bool, careful: bool, careful_stepsize: float
+    ) -> None:
         """
         Initialize this object.
 
         :param debug: If debugging messages are activated.
         :param dry: If servo outputs are not propagated to the physical servos.
-        :param pins: The GPIO pins that will be used.
+        :param careful: Enable careful mode, which slowly steps the servo to its target, instead of going as fast as possible. This decreases current drawn by the motors, which might be necessary for some robots. This is only available for V2 robots.
+        :param careful_stepsize: How much to step each iteration.
         :raises RuntimeError: If GPIOs could not initialize.
         """
         self._debug = debug
         self._dry = dry
-        self._pins = pins
+        self._careful = careful
+        self._careful_stepsize = careful_stepsize
+
+        self._enabled = False
+        self._update_loop_thread = None
 
         if not self._dry:
             self._gpio = pigpio.pi()
             if not self._gpio.connected:
                 raise RuntimeError("Failed to reach pigpio daemon.")
 
-            for pin in self._pins:
+            for pin in self._PINS:
```

Comment:
I did this so we don't need to know what pins the user is interested in at start up(no need to configure). I don't think there's any drawbacks aside from it being slower, but even that I think should be ok. I need to test on a different v1 robot, i think the one i was using is broken because a student used a different one and it owrked fine